app/grandchallenge/documentation/models.py

```diff
@@ -0,0 +1,185 @@
+        return self.title
+
+    def save(self, *args, **kwargs):
+        # when saving for the first time only, put this page last in order
+        if not self.id:
+            # get max value of order for current pages.
+            try:
+                self.order = (
+                    DocPage.objects.aggregate(Max("order"))["order__max"] + 1
+                )
+            except ObjectDoesNotExist:
+                # Use the default
+                pass
+            except TypeError:
+                # Use the default
+                pass
+
+        self.html = clean(self.content)
+
+        super().save(*args, **kwargs)
+
+    def move(self, move):
+        if move == self.UP:
+            mm = DocPage.objects.get(order=self.order - 1)
+            mm.order += 1
+            mm.save()
+            self.order -= 1
+            self.save()
+        elif move == self.DOWN:
+            mm = DocPage.objects.get(order=self.order + 1)
+            mm.order -= 1
+            mm.save()
+            self.order += 1
+            self.save()
+        elif move == self.FIRST:
+            pages = DocPage.objects.all()
+            idx = index(pages, self)
+            pages[idx].order = pages[0].order - 1
+            pages = sorted(pages, key=lambda page: page.order)
+            self.normalize_page_order(pages)
+        elif move == self.LAST:
+            pages = DocPage.objects.all()
+            idx = index(pages, self)
+            pages[idx].order = pages[len(pages) - 1].order + 1
+            pages = sorted(pages, key=lambda page: page.order)
+            self.normalize_page_order(pages)
+
+    @staticmethod
+    def normalize_page_order(pages):
+        """Make sure order in pages Queryset starts at 1 and increments 1 at
+        every page. Saves all pages
+        """
+        for idx, page in enumerate(pages):
+            page.order = idx + 1
+            page.save()
+
+    def get_absolute_url(self):
+        url = reverse("documentation:detail", kwargs={"slug": self.slug},)
+        return url
+
+    @property
+    def next(self):
+        try:
+            next_page = DocPage.objects.filter(order=self.order + 1).get()
+        except ObjectDoesNotExist:
+            next_page = None
+        return next_page
+
+    @property
+    def previous(self):
+        try:
+            previous_page = DocPage.objects.filter(order=self.order - 1).get()
+        except ObjectDoesNotExist:
+            previous_page = None
+        return previous_page
+
+    @property
+    def parent(self):
+        if self.level == 1:
+            parent = None
+        else:
+            if self.level == 2:
+                parent = DocPage.objects.filter(
+                    order__lt=self.order, level=1
+                ).last()
+            elif self.level == 3:
+                parent = DocPage.objects.filter(
+                    order__lt=self.order, level=2
+                ).last()
+        return parent
+
+    @property
+    def children(self):

```

Comment:
This is going to be really inefficient as it goes to the database a lot. I feel now that using a `parent = models.ForeignKey("self", null=True, blank=True, on_delete=???, related_name="children")` would be a better approach, then getting the children would be a case of doing `page.children.all()`.